ext/curl/curl_string_file.c

```diff
@@ -0,0 +1,50 @@
+/*
+   +----------------------------------------------------------------------+
+   | Copyright (c) The PHP Group                                          |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 3.01 of the PHP license,      |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | http://www.php.net/license/3_01.txt                                  |
+   | If you did not receive a copy of the PHP license and are unable to   |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@php.net so we can mail you a copy immediately.               |
+   +----------------------------------------------------------------------+
+ */
+
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
+#include "php.h"
+#include "curl_private.h"
+#include "curl_string_file_arginfo.h"
+
+PHP_CURL_API zend_class_entry *curl_CURLStringFile_class;
+
+ZEND_METHOD(CURLStringFile, __construct)
+{
+	zend_string *data, *postname, *mime = NULL;
+	zval *object;
+
+	object = ZEND_THIS;
+
+	ZEND_PARSE_PARAMETERS_START(2,3)
+		Z_PARAM_STR(data)
+		Z_PARAM_STR(postname)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_STR_OR_NULL(mime)
+	ZEND_PARSE_PARAMETERS_END();
+
+	zend_update_property_stringl(curl_CURLStringFile_class, Z_OBJ_P(object), "data", sizeof("data")-1, ZSTR_VAL(data), ZSTR_LEN(data));
```

Comment:
This will perform a copy of the data. You can use zend_update_property_str to avoid the copy.